### PR TITLE
chore: remove "/sources" endpoint

### LINKED ISSUE #15571: remove `/sources` endpoint from the API
this endpoint is not used and should be removed from the api and swagger.

also, connect https://github.com/influxdata/influxdb/issues/13325

### PR DESCRIPTION
**Pull request by nwneisen**
Signed-off-by: nwneisen <nwneisen@gmail.com>

Closes #15571 

Describe your proposed changes here.
Remove the "/chronograf/v1/sources" endpoint

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass
- [x] http/swagger.yml updated (if modified Go structs or API)
- [ ] Documentation updated or issue created (provide link to issue/pr)
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: chronograf/server/config.go [hunk 1]
```diff
@@ -12,6 +12,10 @@ type configLinks struct {
 	Auth string `json:"auth"` // Auth link to the auth config endpoint
 }
 
+type selfLinks struct {
```
Comment (desa) [inline chronograf/server/config.go]: What's this for?

Comment (nwneisen) [reply chronograf/server/config.go]: This was the only code in `sources.go` that seemed needed so I moved it out. I mean to comment and make sure to bring it to attention during review.

**Comment by desa** [comment]
@nwneisen can you squash these into a single commit and then we can merge it!

**Comment by nwneisen** [comment]
@desa This should be ready to go!

I'm looking through other issues for ways I can help.  If there's any areas you'd recommend focusing on, let me know and I'll take a look.

**Comment by desa** [comment]
@nwneisen really awesome work!

What types of issues are you interested in?

**Comment by nwneisen** [comment]
@desa I'm open to any but more interested in back-end. I found that I like answering questions on SO and learning about areas such as configuration, API, or CLI would help with this.

**Comment by desa** [comment]
Awesome. Some things that would be really great are

https://github.com/influxdata/influxdb/issues/15838
or
https://github.com/influxdata/influxdb/issues/15818

Let me know if there's not enough information in either of these, or if they don't seem interesting.

Also feel free to send me an email at `<my github username>@influxdata.com` at any point and I'd be happy to find a few more issues that would be good. 😃 .

**Comment by nwneisen** [comment]
Sounds great!  Thanks for the suggestions! @desa

**Pull request merged**